nemo_curator/modules/filter.py

```diff
@@ -204,3 +206,161 @@ def __call__(self, dataset: DocumentDataset) -> DocumentDataset:
+        tgt_field: Union[List[str], str] = "tgt",
+        score_field=None,
+        score_type=None,
+        invert=False,
+    ):
+        """
+        A filter object wrapper class for applying bilingual filter objects (such as length ratio, QE filter) on bitext.
+
+        Args:
+          filter_obj: Needs to be a filter that applies to 2 text columns, as is the case in bitext.
+          score_field: The field to which the scores will be written. If None, scores will be immediately discarded after use.
+        """
+        self.filter_obj = filter_obj
+        self.score_field = score_field
+        self.score_type = score_type
+        self.invert = invert
+
+        if type(src_field) == list and type(tgt_field) == list:
+            assert len(src_field) == len(tgt_field), (
+                "The semantics of JointScoreFilter assumes that the information passed for the source and target side should be the same. "
+                + f"Got {len(src_field)} and {len(tgt_field)}, which means you are doing something unintended."
+            )
+        elif not (type(src_field) == str and type(tgt_field) == str):
+            raise ValueError(
+                "The semantics of JointScoreFilter assumes that the information passed for the source and target side should be the same. "
+                "Got two objects of different types, which means you are doing something unintended."
+            )
+        self.src_field = src_field
+        self.tgt_field = tgt_field
+
+    def __call__(self, dataset: ParallelDataset):
+        # Set the metadata for the function calls if provided
+        if self.score_type:
+            meta = (None, self.score_type)
+        else:
+            meta = no_default
+
+        # support multiple fields if supplied
+        fields = []
+        if type(self.src_field) == list and type(self.tgt_field) == list:
+            fields.extend(self.src_field)
+            fields.extend(self.tgt_field)
+        # constructor made sure that both are strings
+        else:
+            fields.append(self.src_field)
+            fields.append(self.tgt_field)
+
+        if is_batched(self.filter_obj.score_document):
+            scores = dataset.df[fields].map_partitions(
+                self.filter_obj.score_document, meta=meta
+            )
+        else:
+            scores = dataset.df[fields].apply(
+                self.filter_obj.score_document, axis=1, meta=meta
+            )
+
+        if self.score_field is not None:
+            dataset.df[self.score_field] = scores
+
+        if is_batched(self.filter_obj.keep_document):
+            bool_mask = scores.map_partitions(
+                self.filter_obj.keep_document, meta=(None, bool)
+            )
+        else:
+            bool_mask = scores.apply(self.filter_obj.keep_document, meta=(None, bool))
+        if self.invert:
+            bool_mask = ~bool_mask
+
+        return ParallelDataset(dataset.df[bool_mask])
+
+
+class ParallelScoreFilter:
```

Comment:
Correct me if I'm wrong, but is this the same as applying two `ScoreFilter`s one after the other and just changing the field they apply to?

Reply:
Yeah.. I just realized that I could have just apply monolingual filter twice, on source and target fields respectively.

I still think it's nice to have a wrapper class that does that for the user, instead of having users do that manually on parallel data, but I could probably simplify the implementation somewhat.

Reply:
If you think it's nice to have a wrapper class then I'm fine keeping it. Do you think you could change the implementation to just internally create and use two `ScoreFilter` modules? At a glance this seems to be possible, but let me know if it's not